raimitigations/dataprocessing/imputer/iterative_imputer.py

```diff
@@ -0,0 +1,292 @@
+    # ignore everything except the message
+    return str(msg) + "\n"
+
+
+warnings.formatwarning = custom_formatwarning
+
+
+class IterativeDataImputer(DataImputer):
+    """
+    Concrete class that imputes missing data of a feature using the other features. It uses a round-robin method of modeling each feature with missing values to be imputed as a function of the other features.
+    This subclass uses the :class:`~sklearn.impute.IterativeImputer` class from :mod:`sklearn` in the background (note that this sklearn class is still in an experimental stage).
+    sklearn.impute.IterativeImputer can only handle numerical data, however, this subclass allows for categorical input by applying ordinal encoding before calling the sklearn class. In order to use this function, use enable_encoder=True, note that encoded columns are not guaranteed to reverse transform if they have imputed values.
+    If you'd like to use a different type of encoding before imputation, consider using the Pipeline class and call your own encoder before calling this subclass for imputation.
+    For more details see:
+    https://scikit-learn.org/stable/modules/generated/sklearn.impute.IterativeImputer.html#
+
+    :param df: pandas data frame that contains the columns to be imputed;
+
+    :param col_impute: a list of the column names or indexes that will be imputed.
+        If None, this parameter will be set automatically as being a list of all
+        columns with any NaN value;
+
+    :param enable_encoder: a boolean flag to allow for applying ordinal encoding of categorical data before applying the IterativeImputer since it only accepts numerical values.
+
+    :param iterative_params: a dict indicating the parameters used by
+        :class:`~sklearn.impute.IterativeImputer`. The dict has the following structure:
+
+            | {
+            |   **'estimator'**:BayesianRidge(),
+            |   **'missing_values'**:np.nan,
+            |   **'sample_posterior'**:False,
+            |   **'max_iter'**:10,
+            |   **'tol'**:1e-3,
+            |   **'n_nearest_features'**:None,
+            |   **'initial_strategy'**:'mean',
+            |   **'imputation_order'**:'ascending',
+            |   **'skip_complete'**:False,
+            |   **'min_value'**:-np.inf,
+            |   **'max_value'**:np.inf,
+            |   **'random_state'**:None
+            | }
+
+        where these are the parameters used by sklearn's IterativeImputer. If None,
+        this dict will be auto-filled as the one above.
+        ``Note: initial_strategy can take one of these values: ['mean', 'median', 'most_frequent', 'constant']``
+
+    :param verbose: indicates whether internal messages should be printed or not.
+    """
+
+    # -----------------------------------
+    def __init__(
+        self,
+        df: Union[pd.DataFrame, np.ndarray] = None,
+        col_impute: list = None,
+        enable_encoder: bool = False,
+        iterative_params: dict = None,
```

Comment:
So adding this as a third option for the `iterative_params: dict` might be tricky and would need more major changes at this point, but, I like the idea so I added a new parameter called `sklearn_obj` : an `sklearn.impute.IterativeImputer` object to use directly. If this parameter is used, `iterative_params` will be overwritten. 
Using `.get_params()`, I can update the `iterative_params` dict to use these values throughout the code, but pass the object directly to `self.imputer` instead of creating one from the dict.
I did the same for the `KNNDataImputer`. I thought about doing the same for `BasicImputer`, but it seems to be a bit trickier given the split of different types of param dicts.

I updated the tutorial notebooks to use this feature in one of the examples and updated the pytests to cover it as well.